### PR TITLE
initial tvos support and demo app target

### PR DESCRIPTION
**Pull request by sherbondy**
(Resubmitting with history squashed into single commit)
Please test!

I don't think I introduced any regressions (tried on both iOS and tvOS targets after making the changes), but xcodeproj files are a bear, so please let me know if I accidentally tweaked something that causes the file to no longer open in an older version of Xcode that you're interested in continuing to support.

GCDWebServer v3.3 landed about a week ago, noticed on Monday, and finally got around to building the demo app target for tvOS.

Reusing the same AppDelegate source code from the original demo app.

Thanks,
Ethan

### CODE REVIEW

**Comment by mfikes** [comment]
Another error I get when attempting to run the Ambly Demo TV target is an inability to link to the Ambly pod, it appears:

```
ld: library not found for -lAmbly
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by sherbondy** [comment]
Eek, dunno how Info-TV.plist got lost in the fray. Will try to resubmit first thing tomorrow.

**Comment by sherbondy** [comment]
Heh, okay, the missing assets are in. I guess I just got trigger-happy yesterday and completely neglected to add the new files to the commit.

**Comment by sherbondy** [comment]
As far as the linker error, if you ever encounter that again, I think a fresh `pod install` might fix that.

**Comment by mfikes** [comment]
@sherbondy The `Info-TV.plist` issue has now been resolved, but I can't get the linking bit to work (I tried `pod install`.)

What I'm doing is getting the patch from this PR (via https://github.com/omcljs/ambly/pull/115.patch) and applying it to a fresh clone, and trying it out. The iOS demo app links and runs but not the tvOS demo app.

File: ObjectiveC/Ambly Demo/Ambly Demo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -154,17 +193,28 @@
...
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+		};
 		ED387D171A602B56007E40A2 /* Resources */ = {
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 				ED387D281A602B56007E40A2 /* Main.storyboard in Resources */,
 				ED387D2D1A602B56007E40A2 /* LaunchScreen.xib in Resources */,
+				AA64223D1C28A8E5002F7170 /* Info-TV.plist in Resources */,
```
Comment (mfikes) [inline ObjectiveC/Ambly Demo/Ambly Demo.xcodeproj/project.pbxproj:217]: `Info-TV.plist` is listed under Build Phases > Copy Bundle Resources for the Ambly Demo target (the non-tvOS target). I bet it causes no issue but is also probably caused by this line here.

**Comment by mfikes** [comment]
I think the root problem with the linking has to do with the target specifiers in the Podfile. If remove the conditional blocks simply specify `platform :tvos, '9.0'`, then things work, and actually start working on my Mac for subsequent attempts without making this change. (This state fulness probably explains why you don't see it.) I've gone to another Mac and I see that it shows the same issue.

Also, I'm seeing that discovery doesn't work until I restart the app (happens to iOS and tvOS), which may be related to GCDWebServer 3.3.

Given this funkiness, I've put your patch in a tvos branch were these subtle issues can be ironed out before fully merging in, but where people can also try things. Your patch is in here https://github.com/omcljs/ambly/commit/c7caf8533737100f086b652c13cc3b7196a5e4c9

I'll close this PR manually.

**Pull request closed without merging**